packages/lesswrong/components/linkPreview/PostLinkPreview.jsx

```diff
@@ -226,9 +226,11 @@ const DefaultPreview = ({classes, href, innerHTML, anchorEl, hover, onsite=false
       </LWPopper>
 
       {onsite ?
-        <Link to={href} dangerouslySetInnerHTML={{__html: innerHTML}} id={id} /> 
-        :
-        <a href={href} dangerouslySetInnerHTML={{__html: innerHTML}} id={id} />}
+          <Link to={href} dangerouslySetInnerHTML={{__html: innerHTML}} id={id}/>
+          :
+          <Components.AnalyticsTracker eventType="link" eventProps={{to: href, externalLink: true}}>
+            <a href={href} dangerouslySetInnerHTML={{__html: innerHTML}} id={id}/>
```

Comment:
Have you verified that this actually works? Because this seems like it wouldn't work.

Reply:
I've tested everything extensively. It works.

Reply:
I'm curious about the intuition that it wouldn't work.

If you look at how the core <Link> component works (react-router-dom), it's pretty much the same thing: a component wrapped around an <a> tag. And the <Analytics Tracker> component is just a wrapping <span> with an onClick handler. So it's just wrapping the <a> tag with a <span>.

Reply:
I predict it only works if the link points at a page which is slow to load.

The problem is that analytics events are debounced, so when you click on this link, it doesn't get sent immediately. But by the time it would be time to send it, you've navigated to another page, so our Javascript will have stopped.

Reply:
Sounds plausible. If we put in, I can see what data comes in and see if I can estimate the reliability. In all my testing it was reliable, but I don't know well it generalizes.